addons/binding/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/discovery/SMAEnergyMeterDiscoveryService.java

```diff
@@ -79,13 +79,13 @@ private synchronized void discover() {
         properties.put(Thing.PROPERTY_VENDOR, "SMA");
         properties.put(Thing.PROPERTY_SERIAL_NUMBER, energyMeter.getSerialNumber());
         ThingUID uid = new ThingUID(THING_TYPE_ENERGY_METER, energyMeter.getSerialNumber());
-        if (uid != null) {
```

Comment:
I just checked again, it's just different than the others it would indeed mean making all the combinations of discovery and logging similar which is out of scope for this PR,